File: app/src/main/java/me/imzack/app/ender/injector/module/TypeDetailPresenterModule.java
package me.imzack.app.ender.injector.module;

import me.imzack.app.ender.view.contract.TypeDetailViewContract;

import dagger.Module;
import dagger.Provides;

@Module
public class TypeDetailPresenterModule {

    private final TypeDetailViewContract mTypeDetailViewContract;
    private final int mTypeListPosition;

    public TypeDetailPresenterModule(TypeDetailViewContract typeDetailViewContract, int typeListPosition) {
        mTypeDetailViewContract = typeDetailViewContract;
        mTypeListPosition = typeListPosition;
    }

    @Provides
    TypeDetailViewContract provideTypeDetailViewContract() {
        return mTypeDetailViewContract;
    }

    @Provides
    int provideTypeListPosition() {
        return mTypeListPosition;
    }
}

File: app/src/main/kotlin/me/imzack/app/end/injector/module/TypeDetailPresenterModule.kt
package me.imzack.app.end.injector.module

import dagger.Module
import dagger.Provides
import me.imzack.app.end.view.contract.TypeDetailViewContract

@Module
class TypeDetailPresenterModule(private val mTypeDetailViewContract: TypeDetailViewContract, private val mTypeListPosition: Int) {

    @Provides
    fun provideTypeDetailViewContract() = mTypeDetailViewContract

    @Provides
    fun provideTypeListPosition() = mTypeListPosition
}
